workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/ComponentDialogSetContent.kt

```diff
@@ -0,0 +1,71 @@
+package com.squareup.workflow1.ui.container
+
+import android.graphics.drawable.ColorDrawable
+import android.util.TypedValue
+import android.view.WindowManager.LayoutParams.FLAG_DIM_BEHIND
+import android.view.WindowManager.LayoutParams.FLAG_NOT_TOUCH_MODAL
+import androidx.activity.ComponentDialog
+import androidx.activity.setViewTreeOnBackPressedDispatcherOwner
+import com.squareup.workflow1.ui.Screen
+import com.squareup.workflow1.ui.ViewEnvironment
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import com.squareup.workflow1.ui.androidx.WorkflowLifecycleOwner
+import com.squareup.workflow1.ui.show
+import com.squareup.workflow1.ui.startShowing
+import com.squareup.workflow1.ui.toViewFactory
+
+/**
+ * Given a [ComponentDialog], wrap it in an [OverlayDialogHolder] that can drive
+ * the Dialog's content via instances of a particular type of [ScreenOverlay].
+ *
+ * Dialogs managed this way are compatible with
+ * [View.setBackHandler][com.squareup.workflow1.ui.setBackHandler],
+ * and honor the [OverlayArea] and [CoveredByModal] values placed in
+ * the [ViewEnvironment] by the standard [BodyAndOverlaysScreen] container.
+ */
+@WorkflowUiExperimentalApi
+public fun <C : Screen, O : ScreenOverlay<C>> ComponentDialog.setContent(
+  overlay: O,
+  environment: ViewEnvironment
+): OverlayDialogHolder<O> {
+  // Note that we always tell Android to make the window non-modal, regardless of our own
+  // notion of its modality. Even a modal dialog should only block events within
+  // the appropriate bounds, but Android makes them block everywhere.
+  requireNotNull(window) { "Expected to find a window for $this." }.addFlags(FLAG_NOT_TOUCH_MODAL)
```

Comment:
We do this here but then we set up more options on the window below after getting teh background color. Why not do it all together?

Reply:
PR encapsulation leak: in a downstream PR the second use of window moves to a separate method.